### PR TITLE
SHD #160 Reckless Gunslinger Unit Implementation

### PR DESCRIPTION
**Pull request by sjmcculloch**
First PR, reckless gunslinger implementation from SHD. 

Hopefully everything looks correct.

### CODE REVIEW

File: test/server/cards/02_SHD/units/RecklessGunslinger.spec.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+describe('Reckless Gunslinger', function () {
+    integration(function (contextRef) {
+        describe('Reckless Gunslinger\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['reckless-gunslinger']
+                    },
+                    player2: {}
```
Comment (AMMayberry1) [inline test/server/cards/02_SHD/units/RecklessGunslinger.spec.ts:10]: nit: can remove the player2 object completely if it's empty

**Comment by AMMayberry1** [comment]
Hi! Thanks for the great PR :) Can you DM me (Veld) in the Karabast discord so I can get you connected with the dev chat?

**Comment by sjmcculloch** [comment]
> Hi! Thanks for the great PR :) Can you DM me (Veld) in the Karabast discord so I can get you connected with the dev chat?

Done!

File: test/server/cards/02_SHD/units/RecklessGunslinger.spec.ts (hunk 1 above)
Comment (sjmcculloch) [reply test/server/cards/02_SHD/units/RecklessGunslinger.spec.ts]: @AMMayberry1 - good pickup, it has been removed.

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**